packages/api/src/helpers/extensionMessaging.ts

```diff
@@ -83,7 +84,19 @@ const serializeMessage = (msg: APIMessages): any => {
     modifiedMsg.payload.amount = JSON.stringify(modifiedMsg.payload.amount);
   }
 
-  if (typeof modifiedMsg.payload?.flags === 'object') {
+  if (
+    'payload' in modifiedMsg &&
+    'limitAmount' in modifiedMsg.payload &&
+    typeof modifiedMsg.payload.limitAmount === 'object'
```

Comment:
You could replace everything in this if by the following code:
```suggestion
    typeof modifiedMsg.payload?.limitAmount === 'object'
```

Reply:
Done